### PR TITLE
Fix pickers interactions in iOS 16 and above

### PR DESCRIPTION
**Pull request by IlyaBausovAkvelon**
## Description:
After release of iOS 16 the problem of interacting with pickers which appear in the input view(when picker appear instead of a keyboard) was introduced. ios-calabash-server became unable to get such elements. So we decided to implement interaction with pickers using DeviceAgent.

## Changes:
- Added implementation of class SetPickerWheelValue. This class is able to set wheel value for any picker which is present on the screen. It uses XCUITest methods to interact with picker's XCUIElement and it's children wheels.
- Added ability to parse and include tree of the children of the elements returned by route "Query". It was added to the Query class.
- In TestApp there was created new screen for testing pickers interactions.

### CODE REVIEW

File: Server/AutomationActions/Gestures/SetPickerWheelValue.m [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    return @[];
+}
+
+- (void)validate {
+    if ([self.query.queryConfiguration isCoordinateQuery]) {
+        return;
+    }
+}
+
++ (Gesture *)executeWithGestureConfiguration:(GestureConfiguration *)gestureConfig query:(Query *)query completion:(CompletionBlock)completion {
```
Comment (DmitriyKirakosyan) [inline Server/AutomationActions/Gestures/SetPickerWheelValue.m:34]: this method never returns Gesture object? Is it ok?

File: TestApp/App/Misc/DatePickersViewController.m [hunk 2]
```diff
@@ -0,0 +1,33 @@
+//
+//  PickerViewController.m
+//  TestApp
+//
+//  Created by Ilya Bausov on 8/7/23.
```
Comment (DmitriyKirakosyan) [inline TestApp/App/Misc/DatePickersViewController.m]: Remove. Annotations should be as in other files. It should be either empty or Microsoft. Check other files.

File: Server/Application/Application.m [hunk 3]
```diff
@@ -210,4 +210,73 @@ + (NSDictionary *)snapshotTree:(XCElementSnapshot *)snapshot {
...
+    XCUIElementQuery *wheelsQuery = [targetPicker descendantsMatchingType:XCUIElementTypePickerWheel];
+    NSArray <XCUIElement *> *pickerWheels = [wheelsQuery allElementsBoundByIndex];
+
+    // Checking is there any wheels were found for selected picker.
+    if ([pickerWheels count] == 0) {
+        message = [NSString stringWithFormat:@"No wheels were found for picker with index %d", pickerIndex];
+        if (error) {
+            *error = [NSError errorWithDomain:CBXWebServerErrorDomain code:1 userInfo:@{NSLocalizedDescriptionKey : message}];
+        }
+        DDLogDebug(@"ERROR: %@", message);
```
Comment (DmitriyKirakosyan) [inline Server/Application/Application.m:263]: Let's be more specific with error log. Imagine we receive an empty string in error message or some generic error message.

Comment (DmitriyKirakosyan) [reply Server/Application/Application.m:263]: Other code parts have same error messages. I'm fine with keeping it as it is.

File: TestApp/App/Misc/PickersViewController.m [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    self.picker = [[UIPickerView alloc] init];
+    self.picker.dataSource = pickerItems;
+    self.pickerField.inputView = self.picker;
+}
+
+/*
+#pragma mark - Navigation
+
+// In a storyboard-based application, you will often want to do a little preparation before navigation
```
Comment (DmitriyKirakosyan) [inline TestApp/App/Misc/PickersViewController.m]: If the code is not needed, let's remove it.

**Review by DmitriyKirakosyan — APPROVED** [review]

File: Server/AutomationActions/Gestures/SetPickerWheelValue.m (hunk 1 above)
Comment (IlyaBausovAkvelon) [reply Server/AutomationActions/Gestures/SetPickerWheelValue.m:34]: This is how Gesture class implemented. I am using Gesture interface to create this class. For now I didn't wanted to create separate class for that.

File: TestApp/App/Misc/PickersViewController.m (hunk 4 above)
Comment (IlyaBausovAkvelon) [reply TestApp/App/Misc/PickersViewController.m]: Fixed.

**Review by DmitriyKirakosyan — APPROVED** [review]

**Pull request merged**